pkg/image/registryclient/client.go

```diff
@@ -369,8 +370,22 @@ func isTemporaryHTTPError(err error) (time.Duration, bool) {
 	switch t := err.(type) {
 	case net.Error:
 		return time.Second, t.Temporary() || t.Timeout()
+	case errcode.ErrorCoder:
+		// note: we explicitly do not check errcode.ErrorCodeUnknown because that is used in
+		// a wide range of scenarios to convey "generic error", not "retryable error"
+		switch {
+		case t.ErrorCode() == errcode.ErrorCodeUnavailable:
+			return 5 * time.Second, true
+		case t.ErrorCode() == errcode.ErrorCodeTooManyRequests:
+			return 2 * time.Second, true
+		}
 	case *registryclient.UnexpectedHTTPResponseError:
-		if t.StatusCode == http.StatusTooManyRequests {
+		switch {
+		case t.StatusCode == http.StatusInternalServerError:
+			return 5 * time.Second, true
+		case t.StatusCode == http.StatusServiceUnavailable:
```

Comment:
Don't we want also retry on StatusBadGateway (502) and StatusGatewayTimeout (504)? These usually indicate temporary server-side problems (when the server is overloaded).

Reply:
Yeah, I think so.  Let me double check the behavior of the daemon in those cases.